composer/trainer/trainer.py

```diff
@@ -1018,9 +1092,182 @@ def saved_checkpoints(self) -> List[Tuple[Timestamp, List[pathlib.Path]]]:
             return []
         return self._checkpoint_saver.saved_checkpoints
 
-    def fit(self):
-        """Train and evaluate the model on the provided data."""
-        # Print any exception, so it can be caputred by any callbacks or loggers (e.g. WandB, FileLogger)
+    def fit(
+        self,
+        *,
+        # Train Dataloader
+        train_dataloader: Optional[Union[Iterable, DataSpec, Dict[str, Any]]] = None,
+        train_dataloader_label: str = "train",
+        train_subset_num_batches: Optional[int] = None,
+        compute_training_metrics: Optional[bool] = None,
+
+        # Timing
+        duration: Optional[Union[int, str, Time[int]]] = None,
+        reset_timer: bool = False,
+
+        # Schedulers
+        schedulers: Optional[Union[ComposerScheduler, PyTorchScheduler, Sequence[Union[ComposerScheduler,
+                                                                                       PyTorchScheduler]]]] = None,
+        scale_schedule_ratio: float = 1.0,
+        step_schedulers_every_batch: Optional[bool] = None,
+
+        # Evaluation
+        eval_dataloader: Optional[Union[Iterable, DataSpec, Evaluator, Sequence[Evaluator]]] = None,
+        eval_subset_num_batches: int = -1,
+        eval_interval: Union[int, str, Time, Callable[[State, Event], bool]] = 1,
+
+        # Numerics
+        grad_accum: Optional[Union[int, str]] = None,
+        precision: Optional[Union[str, Precision]] = None,
+
+        # Grad Clipping
+        grad_clip_norm: Optional[float] = None,
+    ):
+        """Train the model.
+
+        .. note::
+
+            All arguments to :meth:`.fit` are optional. Any values specified here will override
+            what was provided when constructing the :class:`.Trainer`.
```

Comment:
This should not be in a note, but probably in the main body of the docstring.

Should also mention that we support the multiple calls to fit, perhaps with an example. this should also demonstrate how this interacts with the `max_duration` argument, as that can be confusing to folks.